packages/apt/0010-prevent-usage-as-root.patch

```diff
@@ -25,82 +27,22 @@ diff -uNr apt-2.1.11/cmdline/apt-get.cc apt-2.1.11.mod/cmdline/apt-get.cc
 +      return 1;
 +   }
 +
-    // Parse the command line and initialize the package library
     CommandLine CmdL;
-    auto const Cmds = ParseCommandLine(CmdL, APT_CMD::APT_GET, &_config, &_system, argc, argv, &ShowHelp, &GetCommands);
-diff -uNr apt-2.1.11/cmdline/apt-key.in apt-2.1.11.mod/cmdline/apt-key.in
---- apt-2.1.11/cmdline/apt-key.in	2020-11-05 16:50:45.426584174 +0200
-+++ apt-2.1.11.mod/cmdline/apt-key.in	2020-11-05 16:59:59.422661972 +0200
-@@ -83,11 +83,11 @@
-     chmod 700 "$GPGHOMEDIR"
- }
- 
--requires_root() {
--	if [ ! -d "@TERMUX_PREFIX@" ] && [ "$(id -u)" -ne 0 ]; then
--		apt_error "This command can only be used by root."
--		exit 1
--	fi
-+assert_no_root() {
-+    if [ "$(id -u)" = "0" ]; then
-+        apt_error "Ability to run this command as root has been disabled for safety purposes."
-+        exit 1
-+    fi
- }
+    auto const Cmds = ParseCommandLine(CmdL, APT_CMD::APT_MARK, &_config, &_system, argc, argv, &ShowHelp, &GetCommands);
  
- command_available() {
-@@ -624,7 +624,6 @@
- 	 create_new_keyring() { if [ ! -r "$FORCED_KEYRING" ]; then TRUSTEDFILE='/dev/null'; FORCED_KEYRING="$TRUSTEDFILE"; fi; }
- 	 ;;
-       --fakeroot)
--	 requires_root() { true; }
- 	 ;;
-       --quiet)
- 	 aptkey_echo() { true; }
-@@ -762,7 +761,7 @@
- case "$command" in
-     add)
- 	warn_on_script_usage
--	requires_root
-+	assert_no_root
- 	setup_merged_keyring
- 	aptkey_execute "$GPG" --quiet --batch --import "$@"
- 	merge_back_changes
-@@ -771,20 +770,20 @@
-     del|rm|remove)
- 	# no script warning here as removing 'add' usage needs 'del' for cleanup
- 	warn_outside_maintscript
--	requires_root
-+	assert_no_root
- 	foreach_keyring_do 'remove_key_from_keyring' "$@"
- 	aptkey_echo "OK"
-         ;;
-     update)
- 	warn_on_script_usage
--	requires_root
-+	assert_no_root
- 	setup_merged_keyring
- 	update
- 	merge_back_changes
- 	;;
-     net-update)
- 	warn_on_script_usage
--	requires_root
-+	assert_no_root
- 	setup_merged_keyring
- 	net_update
- 	merge_back_changes
-diff -uNr apt-2.1.11/cmdline/apt-mark.cc apt-2.1.11.mod/cmdline/apt-mark.cc
---- apt-2.1.11/cmdline/apt-mark.cc	2020-10-21 12:53:18.000000000 +0300
-+++ apt-2.1.11.mod/cmdline/apt-mark.cc	2020-11-05 16:57:19.205507031 +0200
-@@ -449,6 +449,11 @@
+diff --git a/cmdline/apt.cc b/cmdline/apt.cc
+index af8a95fa5..88c2692ad 100644
+--- a/cmdline/apt.cc
++++ b/cmdline/apt.cc
+@@ -105,6 +105,11 @@ static std::vector<aptDispatchWithHelp> GetCommands()			/*{{{*/
  									/*}}}*/
- int main(int argc,const char *argv[])					/*{{{*/
+ int main(int argc, const char *argv[])					/*{{{*/
  {
 +   if (getuid() == 0) {
 +      std::cout << "Ability to run this command as root has been disabled permanently for safety purposes." << std::endl;
 +      return 1;
 +   }
 +
     CommandLine CmdL;
-    auto const Cmds = ParseCommandLine(CmdL, APT_CMD::APT_MARK, &_config, &_system, argc, argv, &ShowHelp, &GetCommands);
```

Comment:
Probably this part should be preserved as well. It should be checked in more than one place.

Also probably we should enforce repo metadata updating somewhere here, since some people use `apt` instead of `pkg` and they have `404` error while trying to download packages.

Reply:
> Probably this part should be preserved as well. It should be checked in more than one place.

I'm not sure what exactly you are intending to preserve here.

> Also probably we should enforce repo metadata updating somewhere here, since some people use `apt` instead of `pkg` and they have `404` error while trying to download packages.

That'd be a fairly substantial change to the upstream behavior.
I don't think we should change that.
We recommend people to use `pkg`, if they choose to use `apt` directly it is on them to use it properly.

Reply:
Anyway in the case of outdated repo metadata they will get errors. And they are being reported as bugs to all platforms (telegram, reddit, github) pretty much frequently.
But in the case if you do not want to make it less aggressive/insistent you can make longer up-to-date period for metadata.

Reply:
It's out of scope for this PR and I think that's that end of that for now.